### PR TITLE
switch to bulk op and metafieldsSet mutation

### PR DESCRIPTION
**Pull request by tekhaus**
This update _significantly_ improves the manual run time for this task by using a bulk operation to query variants and then the new [metafieldsSet](https://shopify.dev/api/admin-graphql/unstable/mutations/metafieldsset) mutation to apply the metafield updates in batches of 25.

This task requires the 2021-10 Shopify API or greater, so it should not be merged until that is released (expected Oct 1).

### CODE REVIEW

**Comment by tekhaus** [comment]
[Modernize/Update - "Sync inventory levels to variant metafields"](https://trello.com/c/FyMa0W9T/40-modernize-update-sync-inventory-levels-to-variant-metafields)

File: docs/sync-inventory-levels-to-variant-metafields/script.liquid [hunk 1]
```diff
@@ -1,75 +1,205 @@
...
   {% endif %}
+
 {% elsif event.topic == "mechanic/user/trigger" %}
-  {% for variant in shop.variants %}
-    {% for inventory_level in variant.inventory_levels %}
-      {% assign inventory_level_string = inventory_level.location_id | append: "" %}
+  {% capture bulk_operation_query %}
+    query {
+      productVariants(
+        query: ""
```
Comment (isaacbowen) [inline docs/sync-inventory-levels-to-variant-metafields/script.liquid:97]: @tekhaus Is there a purpose to this bit?

Comment (tekhaus) [reply docs/sync-inventory-levels-to-variant-metafields/script.liquid:97]: Ah, yeah, sorry. I was considering adding a query filter option field, and ultimately decided not to, and accidentally left that line in.

Comment (tekhaus) [reply docs/sync-inventory-levels-to-variant-metafields/script.liquid]: Removed in latest commit

**Pull request merged**